app/workers/wp_api_worker.rb

```diff
@@ -9,7 +9,12 @@ class WpApiWorker
   include Sidekiq::Worker
   sidekiq_options :retry => false
 
-  def perform(klass, wp_id)
-    klass.constantize.create_or_update(klass.constantize.wp_type, wp_id)
+  def perform(klass, wp_id, options = {preview: false})
+    # previews have a different endpoint in the json api
+    if options['preview']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.